docs/usage/configuration-options.md

```diff
@@ -421,6 +421,37 @@ If enabled, all issues created by Renovate are set as confidential, even in a pu
 !!! note
     This option is applicable to GitLab only.
 
+## configMigration
+
+If enabled, Renovate will raise a pull request if config file migration is needed.
+
+As Renovate evolves and new feature are added, configuration options might evolve as well.
+For such cases, Renovate will raise config migration pull requests.
+
+Example:
+
+When [`baseBranches`](https://docs.renovatebot.com/configuration-options/#basebranches) was introduced, such pull request would offer the following changes:
+
+From
+
+```json
+{
+  "baseBranch": "main"
+}
+```
+
+To
+
+```json
+{
+  "baseBranches": ["main"]
+}
+```
+
+<!-- prettier-ignore -->
+!!! info
+    This feature writes plain JSON, any JSON5 content will be down leveled and all comments will be removed.
```

Comment:
- it writes plain JSON for `.json` files and JSON5` for `.json` files.
- all comments will be removed.

Reply:
so does "All JSON5 comments will be removed." sums it up accurately?

Reply:
i think yes. but we now support json5 in normal `renovate.json` too, so that will be converted to plain json too.

Reply:
im just cant get my head around - 

> it writes plain JSON for .json files and JSON5for.json` files.

is this want we want?

This feature writes plain JSON for both `.json` and `.json5` files, any content will be down leveled to plain JSON and all comments will be removed.

Reply:
```suggestion
    This feature writes plain JSON for `.json` files, and JSON5 for `.json5` files. So JSON5 content can potentially be down leveled (`.json` files) and all comments will be removed.
```

Something like that?